Library/Homebrew/diagnostic.rb

```diff
@@ -574,16 +574,17 @@ def check_coretap_git_origin
         return if !Utils.git_available? || !(coretap_path/".git").exist?
 
         origin = coretap_path.git_origin
+        default_remote = CoreTap.instance.default_remote
 
         if origin.nil?
           <<~EOS
             Missing #{CoreTap.instance} git origin remote.
 
             Without a correctly configured origin, Homebrew won't update
             properly. You can solve this by adding the Homebrew remote:
-              git -C "#{coretap_path}" remote add origin #{Formatter.url("https://github.com/Homebrew/homebrew-core.git")}
+              git -C "#{coretap_path}" remote add origin #{Formatter.url(default_remote)}
           EOS
-        elsif origin !~ %r{Homebrew/homebrew-core(\.git|/)?$}
+        elsif origin !~ %r{#{URI(default_remote).path}(\.git|/)?$}
```

Comment:
Use `CoreTap.instance.full_name` instead.

Reply:
On Linux, `CoreTap.instance.full_name` returns `Homebrew/homebrew-core` rather than the desired `Linuxbrew/homebrew-core`.

Reply:
I'd like to use…
```ruby
        elsif origin != default_remote
```
It's stricter in three ways:

1. Doesn't allow trailing `.git`
2. Doesn't allow trailing `/`
3. Doesn't allow `ssh`

I'm personally okay with that. Homogeneity is good.

Reply:
Now that I think about it, it's simply a bug that `CoreTap.instance.full_name` returns Homebrew/homebrew-core on Linux. I'll fix it.

Reply:
> Now that I think about it, it's simply a bug that `CoreTap.instance.full_name` returns Homebrew/homebrew-core on Linux. I'll fix it.

👍